dashboard/src/components/ColorSchemeToggle.tsx

```diff
@@ -0,0 +1,50 @@
+import IconButton, { IconButtonProps } from '@mui/joy/IconButton';
+import { useColorScheme } from '@mui/joy/styles';
+import * as React from 'react';
+
+import DarkModeRoundedIcon from '@mui/icons-material/DarkModeRounded';
+import LightModeIcon from '@mui/icons-material/LightMode';
+
+export default function ColorSchemeToggle(props: IconButtonProps) {
+    const { onClick, sx, ...other } = props;
+    const { mode, setMode } = useColorScheme();
+    const [mounted, setMounted] = React.useState(false);
+
+    React.useEffect(() => {
+        setMounted(true);
+    }, []);
+
+    if (!mounted) {
+        return <IconButton size="sm" variant="outlined" color="neutral" {...other} sx={sx} disabled />;
+    }
+
+    return (
+        <IconButton
+            id="toggle-mode"
+            size="sm"
+            variant="soft"
+            color="neutral"
+            {...other}
+            onClick={(event) => {
+                if (mode === 'light') {
```

Comment:
I would create a type called type Mode = {dark = "dark" light = "light"} and use in the code like Mode.light or Mode.dark.